src/charm_state.py

```diff
@@ -586,7 +586,7 @@ def from_charm(cls, charm: CharmBase) -> "OpenstackRunnerConfig":
         )
 
 
-def _build_proxy_config_from_charm() -> "ProxyConfig":
+def build_proxy_config_from_charm() -> "ProxyConfig":
```

Comment:
As we move from externalising the app (so not using as library anymore), I think we should no longer import these classes from the app path. Otherwise we are using the github-runner-manager as library and application at the same time, which could lead to confusion and weird mistakes in the future. The code of the app should be independent from the code of the charm.

WDYT @javierdelapuente ?

I think we should move to something like `ProxyConfig.from_charm` with ProxyConfig living in this module.

Reply:
I will do this in a follow up PR.
Thanks.